examples/fem99.py

```diff
@@ -0,0 +1,99 @@
+import taichi as ti
+try:
+    import taichi_glsl as tl
+except ImportError:
+    print('This example needs the extension library Taichi GLSL to work.'
+          'Please run `pip install --user taichi_glsl` to install it.')
+ti.init(arch=ti.gpu)
+
+N = 32
+dt = 1e-4
+dx = 1 / N
+NF = 2 * N ** 2
+NV = (N + 1) ** 2
+E, nu = 4e4, 0.2
+mu, lam = E / 2 / (1 + nu), E * nu / (1 + nu) / (1 - 2 * nu)
+ball_pos, ball_radius = tl.vec(0.5, 0.0), 0.32
+gravity = tl.vec(0, -40)
+damping = 12.5
+
+pos = ti.Vector.var(2, ti.f32, NV, needs_grad=True)
+vel = ti.Vector.var(2, ti.f32, NV)
+f2v = ti.Vector.var(3, ti.i32, NF)
+B = ti.Matrix.var(2, 2, ti.f32, NF)
+F = ti.Matrix.var(2, 2, ti.f32, NF, needs_grad=True)
+V = ti.var(ti.f32, NF)
+phi = ti.var(ti.f32, NF)
+U = ti.var(ti.f32, (), needs_grad=True)
+
+@ti.kernel
+def update_U():
+    for i in range(NF):
+        ia, ib, ic = f2v[i]
+        a = pos[ia]
+        b = pos[ib]
+        c = pos[ic]
+        V[i] = abs((a - c).cross(b - c))
+        D_i = ti.Matrix.cols([a - c, b - c])
+        F[i] = D_i @ B[i]
+    for i in range(NF):
+        F_i = F[i]
+        J_i = F_i.determinant()
+        log_J_i = ti.log(J_i)
+        phi_i = mu / 2 * ((F_i.transpose() @ F_i).trace() - 2)
+        phi_i -= mu * log_J_i
+        phi_i += lam / 2 * log_J_i ** 2
+        phi[i] = phi_i
+        U[None] += V[i] * phi_i
+
+@ti.kernel
+def advance():
+    for i in range(NV):
+        f_i = -pos.grad[i] * (1 / dx)
```

Comment:
Should `dx` here be the mass of the vertex? `1 / dx` is slightly confusing. Or we can comment that we use `dx` and the mass of the vertex.

Reply:
Yes, so we should use `(m0 / dx ** 2)` instead where `m0` is the total mass?

Reply:
A better solution sounds like `rho * dx ** 2` where `rho` is the area density.